File: mobile/flutter/lib/chess_strategy.dart
class BizXChessStrategy {
  int turn = 0;
  final Map<String, int> resources = {'gold': 100, 'food': 100, 'science': 0};
  int tick(String resource, int amount) { resources[resource] = (resources[resource] ?? 0) + amount; return ++turn; }
  Map<String, String> uci(String fen, {String command = 'go movetime 100'}) => {'protocol': 'UCI', 'fen': fen, 'command': command, 'adapter': 'external-engine'};
}

File: apple/ChessStrategy.swift
import Foundation

public struct BizXChessStrategy {
    public private(set) var turn: Int = 0
    public private(set) var resources: [String:Int] = ["gold":100,"food":100,"science":0]
    public mutating func tick(_ resource: String, _ amount: Int) -> Int { resources[resource, default: 0] += amount; turn += 1; return turn }
    public func uci(_ fen: String, command: String = "go movetime 100") -> [String:String] { ["protocol":"UCI","fen":fen,"command":command,"adapter":"external-engine"] }
}
